### ISSUE TITLE
How does helmfile provider authenticate to kubectl : with external kubeconfig or with kubernetes provider

### ISSUE THREAD

**Issue by abdennour** [mumoshu/terraform-provider-helmfile#20]
I am using kubernetes terraform provider 

```tf
provider "kubernetes" {
  host                   = data.aws_eks_cluster.cluster.endpoint
  cluster_ca_certificate = base64decode(data.aws_eks_cluster.cluster.certificate_authority.0.data)
  token                  = data.aws_eks_cluster_auth.cluster.token
  load_config_file       = false
  version                = "~> 1.9"
}
```

Does helmfile leverage this configuration ? or it just target the cluster with host kubeconfig ? 

On other words:
- Does helmfile ignore the kubernetes provider ? 
- If so, does it rely on the host configuration of KUBECONFIG , AWS_PROFILE ,.. and so on ?

**Comment by mumoshu** [comment]
Hey. There's no way for a Terraform provider to interoperate with other providers. So, it's impossible to read kubernetes provider configuration from the helmfile provider.

For the second question, the provider should use `KUBECONFIG`, which is the same behaviour as helmfile.

**Comment by abdennour** [comment]
Thank you  @mumoshu . I updated my question. please check the  last snippet.. should it work ?

**Comment by mumoshu** [comment]
@abdennour Yeah, it should work as long as long as `module.eks.kubeconfig_filename` points a valid relative/absolute path to a KUBECONFIG. I'm not sure what's `module.eks.kubeconfig_filename` though. Is there any community tf module like that?

**Comment by mumoshu** [comment]
@abdennour I've added `KUBECONFIG` to the second `helmfile_release_set` example in https://github.com/mumoshu/terraform-provider-helmfile#examples.

**Comment by mnatan** [comment]
Hey, can confirm, we are providing `KUBECONFIG` like that and it works.

**Comment by jamesla** [comment]
I just deployed a terraform stack and the helmfile provider deployed to a cluster that was different to what terraform was working on which is a bit dangerous. Lucky for me this was just in a test cluster.

The way that other providers solve this IE the helm provider is via provider settings. This plugin should do the same thing or it's a matter of time until somebody deploys something to the wrong cluster and takes something down in production.

```
provider "helm" {
  version = "1.2"

  kubernetes {
    load_config_file       = false
    host                   = local.k8s_host
    token                  = local.k8s_token
    cluster_ca_certificate = local.k8s_cluster_ca_certificate
  }
}
```

**Comment by mumoshu** [comment]
@jamesla Unfortunately I think that's the problem on terraform and helm provider side, not this provider. AFAIK terraform doesn't provide a way to "source provider config from a tf resource outupt", right?

A simple question that clarifies the where the problem is - does the helm provider supports configuring KUBECONFIG/kubecontext per `resource`, not `provider`? I think the answer is "no". Then you'd need to raise a feature request to the helm provider.

**Comment by mumoshu** [comment]
> helmfile provider deployed to a cluster that was different to what terraform was working on

@jamesla I was thinking that's rather issue in helm provider, as similar as terraform's kuernetes provider. I don't think using provider-level config to configure which KUBECONFIG to use is versatile.

What if you had a single terraform project that creates a K8s cluster AND deploys helmfile releases using the KUBECONFIG generated by the K8s cluster resource? It's impossible when using helm/kubernetes providers as you can't source provider config from resource outputs.

**Comment by lukasmrtvy** [comment]
https://github.com/mumoshu/terraform-provider-helmfile/issues/51 can solve this problem

**Comment by abdennour** [comment]
That's awesome @lukasmrtvy . Did you submit PR ?

**Comment by yashbhutwala** [comment]
I agree with @jamesla, the auth experience of this provider should be consistent with helm and k8s providers.
